### PR TITLE
Use Rest Api to upload to the feed

### PR DESCRIPTION
**Pull request by wli3**
Add pulling logic to make sure it is uploaded to the feed.

Add retry logic for the whole upload process

Remove the old upload script

### CODE REVIEW

**Comment by wli3** [comment]
@rakeshsinghranchi FYI

**Comment by wli3** [comment]
@dotnet-bot test Windows_NT x86 Debug Build

**Comment by wli3** [comment]
@dotnet-bot test CentOS7.1 x64 Debug Build

**Comment by wli3** [comment]
oh hold

**Comment by wli3** [comment]
@dotnet-bot test Windows_NT x86 Debug Build

File: build/publish/PublishDebian.targets [hunk 1]
```diff
@@ -6,28 +6,23 @@
...
-    </DebianUploadJsonContent>
   </PropertyGroup>
 
-  <Target Name="PublishDebFilesToDebianRepo" Condition=" '$(IsDebianBaseDistro)' == 'True' AND '$(SkipPublishToDebianRepo)' != 'true' AND '$(IslinuxPortable)' != 'true' ">
+  <Target Name="PublishDebFilesToDebianRepo"
+          DependsOnTargets="SetupDebProps;"
+          Condition=" '$(IsDebianBaseDistro)' == 'True' AND '$(SkipPublishToDebianRepo)' != 'true' AND '$(IslinuxPortable)' != 'true' " >
     <Error Condition="'$(REPO_ID)' == ''" Text="REPO_ID must be set as an environment variable for debian publishing." />
     <Error Condition="'$(REPO_USER)' == ''" Text="REPO_USER must be set as an environment variable for debian publishing." />
     <Error Condition="'$(REPO_PASS)' == ''" Text="REPO_PASS must be set as an environment variable for debian publishing." />
```
Comment (johnbeisner) [inline build/publish/PublishDebian.targets]: REPO_PASS is not an environment variable - it is set as a 'property'. The error text is misleading.

Comment (wli3) [reply build/publish/PublishDebian.targets]: Thanks, and fixed, now it is required as a MsBuild Property. by environment not is allowed anymore

**Comment by wli3** [comment]
@dotnet-bot test Windows_NT x64 Release Build

**Pull request merged**